server/graphql/v2/mutation/RootMutations.ts

```diff
@@ -49,6 +49,61 @@ const BanAccountResponse = new GraphQLObjectType({
  * Root mutations
  */
 export default {
+  editAccountFlags: {
+    type: new GraphQLNonNull(Account),
+    description: '[Root only] Edits account flags (deleted, banned, archived, trusted host)',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account to change the flags for',
+      },
+      isDeleted: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is soft deleted',
+      },
+      isBanned: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is banned',
+      },
+      isArchived: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is archived',
+      },
+      isTrustedHost: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is a trusted host',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<Record<string, unknown>> {
+      checkRemoteUserCanRoot(req);
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true, paranoid: false });
+
+      if (args.isDeleted && !account.deletedAt) {
+        await account.destroy();
+      } else if (!args.isDeleted && account.deletedAt) {
+        await account.restore();
+
+        // We have freed the slug, so we have to restore it to its original value
+        const timestampPart = account.slug.match(/-\d+$/)[0];
+        await account.update({ slug: account.slug.replace(timestampPart, '') });
+      }
+
+      if (args.isArchived && !account.deactivatedAt) {
+        await account.update({ deactivatedAt: Date.now() });
+      } else if (!args.isArchived && account.deactivatedAt) {
+        await account.update({ deactivatedAt: null });
+      }
```

Comment:
:no_entry_sign: That means that, if I call this mutation on an archived account:

```gql
editAccountFlags(account, { isTrustedHost: false }) { ... }
```

it will unarchive the account, since `args.isArchived` will be falsy (`!undefined === true`). Let's make this one and the others around a bit more flexible:

```suggestion
      } else if (args.isArchived === false && account.deactivatedAt) {
        await account.update({ deactivatedAt: null });
      }
```

Reply:
Yes I've corrected this and the other condition related to trusted host to check explicitly for `isArchived` and `isTrustedHost`